repairnator/repairnator-dockerpool/src/main/java/fr/inria/spirals/repairnator/dockerpool/RunnablePipelineContainer.java

```diff
@@ -46,6 +50,59 @@ public RunnablePipelineContainer(AbstractPoolManager poolManager, String imageId
         this.treatedBuildTracking = treatedBuildTracking;
         this.skipDelete = skipDelete;
         this.createOutputDir = createOutputDir;
+
+        this.containerName = "repairnator-pipeline_"+ Utils.formatFilenameDate(new Date())+"_"+this.buildId;
+        this.output = (createOutputDir) ? "/var/log/"+this.repairnatorConfig.getRunId() : "/var/log";
+
+        this.envValues = new String[] {
+            "BUILD_ID="+this.buildId,
+            "LOG_FILENAME="+containerName,
+            "GITHUB_LOGIN="+System.getenv("GITHUB_LOGIN"),
+            "GITHUB_OAUTH="+System.getenv("GITHUB_OAUTH"),
+            "RUN_ID="+this.repairnatorConfig.getRunId(),
+            "GOOGLE_ACCESS_TOKEN="+this.repairnatorConfig.getGoogleAccessToken(),
+            "REPAIR_MODE="+this.repairnatorConfig.getLauncherMode().name().toLowerCase(),
+            "SPREADSHEET_ID="+this.repairnatorConfig.getSpreadsheetId(),
+            "PUSH_URL="+this.repairnatorConfig.getPushRemoteRepo(),
+            "MONGODB_HOST="+this.repairnatorConfig.getMongodbHost(),
+            "MONGODB_NAME="+this.repairnatorConfig.getMongodbName(),
+            "SMTP_SERVER="+this.repairnatorConfig.getSmtpServer(),
+            "NOTIFY_TO="+ StringUtils.join(this.repairnatorConfig.getNotifyTo(),','),
+            "OUTPUT="+output
+        };
+    }
+
+    public RunnablePipelineContainer(AbstractPoolManager poolManager, String imageId, int buildId, int nextBuildId, String logDirectory, TreatedBuildTracking treatedBuildTracking, boolean skipDelete, boolean createOutputDir) {
+        this.poolManager = poolManager;
```

Comment:
There is lot of duplicated code between the first and the second constructor. I propose you to call the first constructor and change the array to add the nextBuildId variable. It will be easier to maintain.

Reply:
Agreed and done.